### ISSUE TITLE
Does not work with `jsarticle`, `jsbook`.

### ISSUE THREAD

**Issue by ryota2357** [latex-lsp/texlab#793]
I'm using neovim and nvim-lsp.

As in the title, texlab will detach (clash?) with `documentclass{jsarticle}`. `article` and `jarticle` works well.

## How to reproduce

Use this init.vim.

<details>
<summary>init.vim</summary>

```vim
if has('vim_starting')
  set encoding=utf-8
endif
scriptencoding utf-8

if &compatible
  set nocompatible
endif

let s:plug_dir = expand('/tmp/plugged/vim-plug')
if !filereadable(s:plug_dir .. '/plug.vim')
  execute printf('!curl -fLo %s/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim', s:plug_dir)
end

execute 'set runtimepath+=' . s:plug_dir
call plug#begin(s:plug_dir)
Plug 'hrsh7th/nvim-cmp'
Plug 'hrsh7th/cmp-buffer'
Plug 'hrsh7th/cmp-nvim-lsp'
Plug 'hrsh7th/vim-vsnip'
Plug 'neovim/nvim-lspconfig'
call plug#end()
PlugInstall | quit

let g:tex_flavor = 'latex'

lua << EOF
local cmp = require "cmp"
cmp.setup {
  snippet = {
    expand = function(args)
      vim.fn["vsnip#anonymous"](args.body)
    end,
  },
  mapping = cmp.mapping.preset.insert({
    ['<CR>'] = cmp.mapping.confirm({ select = true }),
  }),
  sources = cmp.config.sources({
    { name = "nvim_lsp" },
    { name = "buffer" },
  }),
}
EOF

lua << EOF
local capabilities = require('cmp_nvim_lsp').default_capabilities()
require('lspconfig').texlab.setup {
    capabilities = capabilities
}
EOF
```
</details>

1. Start neovim.
2. `:LspInfo`, you can see that texlab is started.
3. Type `\documentclass{jsarticle}` in the first line. (You can also get completion.)
4. Go to the next line, and type `\begin`. (You can't get completion.)
5. `:LspInfo`, you can see that texlab is detached.

Also, the diagnostic does not work well.

Open the following tex file.

```tex
\documentclass{jarticle}
\begin{document}
\begin{equation}
  x^2 + y^2 = z^2
\end{equation}
\end{document}
```

1. Change `\begin{equation}` to `\begin{equation` (remove `}`)
2. You can see the diagnostic error.
3. Continue to write something and save it. (don't fix `}`)
4. You can't see the error.

Note: In this init.vim, I used nvim-cmp for completion. The same thing is occurs in ddc.vim

## Expected

If you do the above with `jarticle` or `article` instead of `jsarticle`, the input completion and errors will continue to display correctly.

## Environment

neovim

```
:version
NVIM v0.8.0
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@HMBRW-A-001-M1-004.local
```

texlab

```
$ texlab --version
texlab 4.3.0
```

If this bug originates from nevoim or nvim-lsp, I apologize.

**Comment by ryota2357** [comment]
In VSCode, the same(?) things occur. I installed texlab extension. (not installed LaTeX Workshop.)

With `jsarticle`, TexLab language server output is this.

```
[Info  - 15:12:47] Connection to server got closed. Server will restart.
[Info  - 15:12:50] Connection to server got closed. Server will restart.
[Info  - 15:12:54] Connection to server got closed. Server will restart.
[Info  - 15:13:07] Connection to server got closed. Server will restart.
[Error - 15:13:08] The texlab server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
```

This is not reproduced with `jarticle` and so on.

**Comment by pfoerster** [comment]
@ryota2357 Thanks for the detailed report! Can you give 7fd82cc5885fb6e180823ce7715eb173283facfe a try? It should fix the issue.

**Comment by ryota2357** [comment]
@pfoerster
Today, I tested 7fd82cc5885fb6e180823ce7715eb173283facfe in other PC, it dose not fix the issue.

<details>
<summary>vimrc-dev (almost same as first init.vim)</summary>

```vim
if has('vim_starting')
  set encoding=utf-8
endif
scriptencoding utf-8

if &compatible
  set nocompatible
endif

let s:plug_dir = expand('/tmp/plugged/vim-plug')
if !filereadable(s:plug_dir .. '/plug.vim')
  execute printf('!curl -fLo %s/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim', s:plug_dir)
end

execute 'set runtimepath+=' . s:plug_dir
call plug#begin(s:plug_dir)
Plug 'hrsh7th/nvim-cmp'
Plug 'hrsh7th/cmp-nvim-lsp'
Plug 'hrsh7th/vim-vsnip'
Plug 'neovim/nvim-lspconfig'
call plug#end()
PlugInstall | quit

let g:tex_flavor = 'latex'

lua << EOF
local cmp = require "cmp"
cmp.setup {
  snippet = {
    expand = function(args)
      vim.fn["vsnip#anonymous"](args.body)
    end,
  },
  mapping = cmp.mapping.preset.insert({
    ['<CR>'] = cmp.mapping.confirm({ select = true }),
  }),
  sources = cmp.config.sources({
    { name = "nvim_lsp" },
  }),
}
EOF

lua << EOF
local capabilities = require('cmp_nvim_lsp').default_capabilities()
require('lspconfig').texlab.setup {
    capabilities = capabilities
}

vim.lsp.set_log_level("debug")
EOF
```
</details>

https://user-images.githubusercontent.com/61523777/197142832-258bbe1a-0d51-467c-a709-4a50276e522d.mov

neovim version is same(0.8), macOS version is same(12.6).
This PC is MacBook Pro 2018 (intel i5).
But in MacBook Pro 2021 (M1-Pro), this issue is fixed.

**Comment by ryota2357** [comment]
I want to fix the issue, but I don't have skills.
Waht information should I provide?

**Comment by pfoerster** [comment]
@ryota2357
> What information should I provide?

Thanks to your video, I was able to narrow down the issue. Can you try e353156, please?

**Comment by ryota2357** [comment]
Thank you!
It works on both PC.

**Comment by pfoerster** [comment]
Released with `4.3.1`
